test/dedup-id/src/test/test.js

```
@@ -0,0 +1,207 @@
+
+  const authHeader = (token) => {
+    return token ?
+      {
+        headers: {
+          authorization: token()
+        }
+      } :
+      {};
+  };
+
+  const buildUsageDoc = (orgID, dedupId) => {
+    const usageDoc = {
+      start: timestamp,
+      end: timestamp,
+      organization_id: orgID,
+      space_id: 'test-dedup-id-space-id',
+      consumer_id: 'test-dedup-id-consumer-id',
+      resource_id: 'object-storage',
+      plan_id: 'basic',
+      resource_instance_id: 'test-dedup-id-resource-instance-id',
+      measured_usage: [{
+        measure: 'heavy_api_calls',
+        quantity: testQuantity
+      }]
+    };
+
+    if (dedupId)
+      extend(usageDoc, {
+        dedup_id: dedupId
+      });
+
+    return usageDoc;
+  };
+
+  // On local environment GET by location header is not routed to meter
+  const buildCorrectLocationHeaderUrl = (url) => {
+    if(url.indexOf('localhost') > -1)
+      return localMeterURL + url.substring(url.indexOf('v1/') - 1, url.length);
+    return url;
+  };
+
+  const sleep = (duration) => {
+    return new Promise((cb) => setTimeout(cb, duration));
+  };
+
+  const eventually = async (func) => {
+    while (true) {
+      try {
+        return await func();
+      } catch (e) {
+        debug('not ready yet: %o', e.message);
+      }
+      await sleep(pollInterval);
+    }
+  };
+
+  const sendUsage = async (usage) => {
+    const resp = await doPost(collectorURL + '/v1/metering/collected/usage',
+      extend({ body: usage }, authHeader(systemToken)));
+
+    expect(resp.statusCode).to.equal(202);
+    return buildCorrectLocationHeaderUrl(resp.headers.location);
+  };
+
+  before((done) => {
+    if(!secured())
+      done();
+    else {
+      systemToken = oauth.cache(authServerURL, systemClientId, systemClientSecret,
+        'abacus.usage.read abacus.usage.write'
+      );
+
+      systemToken.start((err) => {
+        if (err) debug('Unable to obtain system oAuth token due to %o', err);
```

You should throw an Error here! No need to have the test continue execution if Abacus is secured and you have not managed to obtain a token.
You could have also considered a way to use async by converting `start` to Promise, if possible.